internal/provider/network/routed_resource.go

```diff
@@ -0,0 +1,540 @@
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *networkRoutedResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var (
+		plan *networkRoutedResourceModel
+	)
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	org, _, err := r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving VDC", err.Error())
+		return
+	}
+
+	parentEdgeGatewayOwnerID, errGet := getParentEdgeGatewayID(org, plan.EdgeGatewayID.ValueString())
+	if errGet != nil {
+		resp.Diagnostics.AddError(errGet.Summary, errGet.Detail)
+		return
+	}
+
+	if govcd.OwnerIsVdcGroup(*parentEdgeGatewayOwnerID) {
+		networkMutexKV.KvLock(ctx, *parentEdgeGatewayOwnerID)
+		defer networkMutexKV.KvUnlock(ctx, *parentEdgeGatewayOwnerID)
+	} else {
+		networkMutexKV.KvLock(ctx, plan.EdgeGatewayID.ValueString())
+		defer networkMutexKV.KvUnlock(ctx, plan.EdgeGatewayID.ValueString())
+	}
+
+	ipPool := []staticIPPool{}
+	diags := plan.StaticIPPool.ElementsAs(ctx, &ipPool, true)
+	resp.Diagnostics.Append(diags...)
+
+	orgVDCNetworkConfig := &govcdtypes.OpenApiOrgVdcNetwork{
+		Name:        plan.Name.ValueString(),
+		Description: plan.Description.ValueString(),
+		OwnerRef:    &govcdtypes.OpenApiReference{ID: *parentEdgeGatewayOwnerID},
+
+		NetworkType: govcdtypes.OrgVdcNetworkTypeRouted,
+
+		// Connection is used for "routed" network
+		Connection: &govcdtypes.Connection{
+			RouterRef: govcdtypes.OpenApiReference{
+				ID: plan.EdgeGatewayID.ValueString(),
+			},
+			// API requires interface type in upper case, but we accept any case
+			ConnectionType: plan.InterfaceType.ValueString(),
+		},
+		Subnets: govcdtypes.OrgVdcNetworkSubnets{
+			Values: []govcdtypes.OrgVdcNetworkSubnetValues{
+				{
+					Gateway:      plan.Gateway.ValueString(),
+					PrefixLength: int(plan.PrefixLength.ValueInt64()),
+					DNSServer1:   plan.DNS1.ValueString(),
+					DNSServer2:   plan.DNS2.ValueString(),
+					DNSSuffix:    plan.DNSSuffix.ValueString(),
+					IPRanges: govcdtypes.OrgVdcNetworkSubnetIPRanges{
+						Values: processIPRanges(ipPool),
+					},
+				},
+			},
+		},
+	}
+
+	orgNetwork, err := org.CreateOpenApiOrgVdcNetwork(orgVDCNetworkConfig)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating routing network", err.Error())
+		return
+	}
+
+	plan = &networkRoutedResourceModel{
+		ID:            types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID),
+		Name:          plan.Name,
+		Description:   plan.Description,
+		EdgeGatewayID: plan.EdgeGatewayID,
+		InterfaceType: plan.InterfaceType,
+		Gateway:       plan.Gateway,
+		PrefixLength:  plan.PrefixLength,
+		DNS1:          plan.DNS1,
+		DNS2:          plan.DNS2,
+		DNSSuffix:     plan.DNSSuffix,
+		StaticIPPool:  plan.StaticIPPool,
+	}
```

Comment:
It's not sufficient ?

```suggestion
	plan.ID = types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID)
```